sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchClientBuilder.java

```diff
@@ -102,6 +116,27 @@ public BlobBatchClient buildClient() {
      */
     public BlobBatchAsyncClient buildAsyncClient() {
         BlobServiceVersion serviceVersion = version != null ? version : BlobServiceVersion.getLatest();
-        return new BlobBatchAsyncClient(clientUrl, pipeline, serviceVersion, containerScoped);
+        return new BlobBatchAsyncClient(clientUrl, addBlobUserAgentModificationPolicy(pipeline), serviceVersion,
+            containerScoped);
+    }
+
+    private HttpPipeline addBlobUserAgentModificationPolicy(HttpPipeline pipeline) {
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        for (int i = 0; i < pipeline.getPolicyCount(); i++) {
+            HttpPipelinePolicy currPolicy = pipeline.getPolicy(i);
+            policies.add(currPolicy);
+            if (currPolicy instanceof UserAgentPolicy) {
+                String clientName = PROPERTIES.getOrDefault(SDK_NAME, "UnknownName");
+                String clientVersion = PROPERTIES.getOrDefault(SDK_VERSION, "UnknownVersion");
```

Comment:
The properties should never change during runtime, we can make these values constants

Reply:
Is there an easy way to make it update every time we do a version bump?

I chose properties files since that is what we are already doing for the blob library, file share library, etc

Reply:
@alzimmermsft Updated